File: Projeto_6/src/models/UsuarioAutenticavel.py
from src.models.Usuario import Usuario
from abc import ABC, abstractmethod
from src.models.TipoUsuario import TipoUsuario

class UsuarioAutenticavel (Usuario, ABC):
    def __init__(self, nome='indefinido', email='indefinido', tipo=TipoUsuario.DEFAULT,senha=''):
        # Chamando o construtor da classe Usuario
        super().__init__(nome, email, tipo)

        self._senha = senha

    @abstractmethod
    def autenticar(self, senha) -> bool:
        """
        Método abstrato que será implementado nas subclasses.
        Verifica se a senha passada é válida.
        """
        pass

File: Projeto_6/Projeto-6-guilherme/src/UsuarioAutenticavel.cpp
#include "UsuarioAutenticavel.h"

UsuarioAutenticavel::UsuarioAutenticavel()
    : Usuario(), senha("") {}

UsuarioAutenticavel::UsuarioAutenticavel(std::string nome, std::string email, std::string tipo, std::string senha)
    : Usuario(nome, email, tipo), senha(senha) {}